tmt/cli.py

```diff
@@ -569,14 +569,30 @@ def export(context, format_, nitrate, bugzilla, **kwargs):
     if bugzilla and not nitrate:
         raise tmt.utils.GeneralError(
             "The --bugzilla option is supported only with --nitrate for now.")
-    for test in context.obj.tree.tests():
-        if nitrate:
+
+    if nitrate:
+        for test in context.obj.tree.tests():
             test.export(format_='nitrate')
+    elif format_ == 'execute':
+        for test in context.obj.tree.tests():
+            echo(test.export(format_='execute'), nl=False)
+    elif format_ in ['dict', 'yaml']:
+        keys = None
+        if kwargs.get('fmf_id'):
+            keys = 'fmf-id'
+
+        data = list()
+        for test in context.obj.tree.tests():
+            test_info = test.export(format_=format_, keys=keys)
+            data.append(test_info)
+
+        if format_ == 'dict':
+            echo(data, nl=False)
         else:
-            if kwargs.get('fmf_id'):
-                echo(test.export(format_=format_, keys='fmf-id'), nl=False)
-            else:
-                echo(test.export(format_=format_), nl=False)
+            echo(tmt.utils.dict_to_yaml(data), nl=False)
+    else:
+        raise tmt.utils.GeneralError(
+            f"Invalid test export format '{format_}'.")
```

Comment:
I'd suggest to use the same approach as for `tmt plan export` below:

https://github.com/teemtee/tmt/blob/ec2e296a88bceba9eed5f14667cc45ad5d6dc5fc/tmt/cli.py#L720

Fetch all test data using `dict` format and then send to the user `dict` or `yaml`. I believe the implementation should then be much more straightforward.

Reply:
Yes, updated.